api/src/main/java/com/github/retrooper/packetevents/util/VectorMutable3i.java

```diff
@@ -0,0 +1,291 @@
+    }
+
+    public VectorMutable3i(long val) {
+        this(val, PacketEvents.getAPI().getServerManager().getVersion());
+    }
+
+    public VectorMutable3i(long val, ServerVersion serverVersion) {
+        int x = (int) (val >> 38);
+        int y;
+        int z;
+
+        // 1.14 method for this is storing X Z Y
+        // 1.17 added support for negative values
+        if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_14)) {
+            y = (int) (val << 52 >> 52);
+            z = (int) (val << 26 >> 38);
+        } else {
+            // 1.13 and below store X Y Z
+            y = (int) ((val >> 26) & 0xFFF);
+            z = (int) (val << 38 >> 38);
+        }
+
+        this.x = x;
+        this.y = y;
+        this.z = z;
+    }
+
+    /**
+     * Constructor allowing you to set the values.
+     *
+     * @param x X
+     * @param y Y
+     * @param z Z
+     */
+    public VectorMutable3i(int x, int y, int z) {
+        this.x = x;
+        this.y = y;
+        this.z = z;
+    }
+
+    /**
+     * Constructor allowing you to specify an array.
+     * X will be set to the first index of an array(if it exists, otherwise 0).
+     * Y will be set to the second index of an array(if it exists, otherwise 0).
+     * Z will be set to the third index of an array(if it exists, otherwise 0).
+     *
+     * @param array Array.
+     */
+    public VectorMutable3i(int[] array) {
+        if (array.length > 0) {
+            x = array[0];
+        } else {
+            x = 0;
+            y = 0;
+            z = 0;
+            return;
+        }
+        if (array.length > 1) {
+            y = array[1];
+        } else {
+            y = 0;
+            z = 0;
+            return;
+        }
+        if (array.length > 2) {
+            z = array[2];
+        } else {
+            z = 0;
+        }
+    }
+
+    public long getSerializedPosition(ServerVersion serverVersion) {
+        // 1.17 adds support for negative values
+        if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_17)) {
+            long x = getX() & 0x3FFFFFF;
+            long y = getY() & 0xFFF;
+            long z = getZ() & 0x3FFFFFF;
+
+            return x << 38 | z << 12 | y;
+        }
+        // 1.14 method for this is storing X Z Y
+        if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_14)) {
+            return ((long) (getX() & 0x3FFFFFF) << 38) | ((long) (getZ() & 0x3FFFFFF) << 12) | (getY() & 0xFFF);
+        }
+        // 1.13 and below store X Y Z
+        return ((long) (getX() & 0x3FFFFFF) << 38) | ((long) (getY() & 0xFFF) << 26) | (getZ() & 0x3FFFFFF);
+    }
+
+    public long getSerializedPosition() {
+        return getSerializedPosition(PacketEvents.getAPI().getServerManager().getVersion());
+    }
+
+    public int getX() {
+        return x;
+    }
+
+    public int getY() {
+        return y;
+    }
+
+    public int getZ() {
+        return z;
+    }
+
+    public boolean equals(VectorMutable3i other) {
```

Comment:
Shouldnt the parameter be the interface? (Same for every other method in this class)